/*
 * Copyright (c) 2020 Huawei Technologies Co.,Ltd.
 *
 * openGauss is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *
 *          http://license.coscl.org.cn/MulanPSL2
 *
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 *---------------------------------------------------------------------------------------
 *
 * IDENTIFICATION
 *        src/gausskernel/cbb/grpc/remote_read.proto
 *
 * ---------------------------------------------------------------------------------------
 */
syntax = "proto3";

package gauss;

service RemoteRead {
	rpc GetCU(CURequest) returns (CUResponse) {}
	rpc GetPage(PageRequest) returns (PageResponse) {}
}

message CURequest {
	uint32 spcnode = 1;
	uint32 dbnode = 2;
	uint32 relnode = 3;
	int32 bucketnode = 4;
	int32 colid = 5;
	uint64 offset = 6;
	int32 size = 7;
	uint64 lsn = 8;
}

message CUResponse {
	int32 return_code = 1;
	int32 size = 2;
	bytes cudata = 3;
}

message PageRequest {
	uint32 spcnode = 1;
	uint32 dbnode = 2;
	uint32 relnode = 3;
	int32 bucketnode = 4;
	int32 forknum = 5;
	uint32 blocknum = 6;
	uint32 blocksize = 7;
	uint64 lsn = 8;
}

message PageResponse {
	int32 return_code = 1;
	int32 size = 2;
	bytes pagedata = 3;
}
